DOC-8878 Enhance Essential Metrics with Alert guidance (#18537)
cockroachdb/docs: src/current/_includes/v24.1/essential-alerts.md

**Rule**
<br>WARNING:  `sys.host.disk.iopsinprogress` greater than `10` for `10 seconds`
<br>CRITICAL:  `sys.host.disk.iopsinprogress` greater than `20` for `10 seconds`

**Action**

- Provision enough storage capacity for CockroachDB data, and configure your volumes to maximize disk I/O. Refer to [Storage and disk I/O]({% link {{ page.version.version }}/common-issues-to-monitor.md %}#storage-and-disk-i-o).

### Version mismatch

All CockroachDB cluster nodes should be running the same exact executable (with identical build label). This warning guards against an operational error where some nodes were not upgraded.

**Metric**
<br>`build.timestamp`

**Rule**
<br>Set alerts for each node:
<br>WARNING:  `build.timestamp` not the same across cluster nodes for more than `4 hours`

**Action**

- Ensure all cluster nodes are running exactly the same CockroachDB version, including the patch release version number.

### High open file descriptor count

Send an alert when a cluster is getting close to the open file descriptor limit.

**Metric**
<br>`sys.fd.open`

> <br>`sys.fd.softlimit`

Reviewer: Some metrics in the "**Metric**" sections do, and some don't.

### SQL queries experiencing high latency

Send an alert when the query latency exceeds a user-determined threshold based on their application’s SLA.

**Metric**
<br>[`sql.service.latency`]({% link {{ page.version.version }}/essential-metrics-self-hosted.md %}#sql-service-latency)
<br>[`sql.conn.latency`]({% link {{ page.version.version }}/essential-metrics-self-hosted.md %}#sql-conn-latency)

**Rule**
<br>WARNING:  (p99 or p90 of `sql.service.latency` plus average of `sql.conn.latency`) is greater than a threshold (based on the user’s application SLA)

**Action**

- Apply the time range of the alert to the [**SQL Activity** pages]({% link {{ page.version.version }}/monitoring-and-alerting.md %}#sql-activity-pages) to investigate. Use the [**Statements** page]({% link {{ page.version.version }}/ui-statements-page.md %}) P90 Latency and P99 latency columns to correlate [statement fingerprints]({% link {{ page.version.version }}/ui-statements-page.md %}#sql-statement-fingerprints) with this alert.

> {% if include.deployment == 'self-hosted' %}

Reviewer 2: LGTM! @mgartner do you have any thoughts/concerns on highlighting these as essential metrics to monitor for SQL?

Author: Hi @mgartner , when you have a chance, would you be able to review this section: [SQL queries experiencing high latency](https://deploy-preview-18537--cockroachdb-docs.netlify.app/docs/v24.2/essential-alerts-self-hosted.html#sql-queries-experiencing-high-latency)?

    {% include_cached copy-clipboard.html %}
    ```sql
    select job_id, status,((high_water_timestamp/1000000000)::int::timestamp)-now() as "changefeed latency",created, left(description,60),high_water_timestamp from crdb_internal.jobs where job_type = 'CHANGEFEED' and status in ('running', 'paused','pause-requested') order by created desc;
    ```

2. Copy the `job_id` for the changefeed job with highest `changefeed latency` and pause the job:

    {% include_cached copy-clipboard.html %}
    ```sql
    PAUSE JOB 681491311976841286;
    ```

3. Check the status of the pause request by running the query from step 1. If the job status is `pause-requested`, check again in a few minutes.

4. After the job is `paused`, resume the job.

    {% include_cached copy-clipboard.html %}
    ```sql
    RESUME JOB 681491311976841286;
    ```

> 5. If the changefeed latency does not progress after the above steps due to lack of cluster resources or availability of the changefeed sink, [contact Support](https://support.cockroachlabs.com).

Reviewer: ```suggestion
5. If the changefeed latency does not progress after these steps due to lack of cluster resources or availability of the changefeed sink, [contact Support](https://support.cockroachlabs.com).

```

### Changefeed has been paused a long time 

> Changefeed jobs should not be paused for a long time because the protected timestamp prevents garbage collections. As a hedge against an operational error, this alert guard against an inadvertently "forgotten" pause. 

Reviewer: I think it would be helpful to link to here: https://www.cockroachlabs.com/docs/stable/protect-changefeed-data

> Changefeed jobs should not be paused for a long time because the protected timestamp prevents garbage collections. As a hedge against an operational error, this alert guard against an inadvertently "forgotten" pause. 

Reviewer: ```suggestion
Changefeed jobs should not be paused for a long time because the protected timestamp prevents garbage collection. To protect from an operational error, this alert guards against an inadvertently forgotten pause. 

```

I think you can drop the quotes here. Optional nit: I think "protect" is quicker to parse here and possibly easier for localization efforts.

**Metric**
<br>[`jobs.changefeed.currently_paused`]({% link {{ page.version.version }}/essential-metrics-{{ include.deployment }}.md %}#changefeed-currently-paused)

**Rule**
<br>WARNING:  `jobs.changefeed.currently_paused` is greater than `0` for more than `15 minutes`
<br>CRITICAL:  `jobs.changefeed.currently_paused` is greater than `0` for more than `60 minutes`

**Action**

1. Check the status of each changefeed using the following SQL query: 

    {% include_cached copy-clipboard.html %}

>     ```sql

Reviewer: SQL style

    select job_id, status,((high_water_timestamp/1000000000)::int::timestamp)-now() as "changefeed latency",created, left(description,60),high_water_timestamp from crdb_internal.jobs where job_type = 'CHANGEFEED' and status in ('running', 'paused','pause-requested') order by created desc;
    ```

2. If all the changefeeds have status as `running`, one or more changefeeds may have run into an error and recovered. In the DB Console, navigate to **Metrics**, [**Changefeeds** dashboard]({% link {{ page.version.version }}/ui-cdc-dashboard.md %}) for the cluster and check the [**Changefeed Restarts** graph]({% link {{ page.version.version }}/ui-cdc-dashboard.md %}#changefeed-restarts).

3. Resume paused changefeed(s) with the `job_id` using:
   
    {% include_cached copy-clipboard.html %}
    ```sql
    RESUME JOB 681491311976841286;
    ```

### Changefeed experiencing high latency

> Send an alert when the latency of any changefeed running on any node is higher than the set threshold, which depends on the [`gc.ttlseconds`]({% link {{ page.version.version }}/configure-replication-zones.md %}#replication-zone-variables) variable set in the cluster. This alert will ensure that the changefeed is progressing quicker than the garbage collection ttl. 

Author: Thanks! updated as suggested
